cmds/dhclient/dhclient.go

```diff
@@ -247,12 +281,17 @@ func main() {
 			} else {
 				done <- dhclient6(iface, *renewals, timeout, *retry)
 			}
-		}(i)
+			debug("Done dhclient for %v", ifname)
+		}(i.Attrs().Name)
 	}
 
+	go func() {
+		wg.Wait()
```

Comment:
I prefer the waitgroup, for that very reason (not having to know how many goroutines were launched). I don't see the cost of adding a goroutine as that big a deal -- heck, we're supposed to treat them as cheap. Since at some point I may want to do both v4 and v6 at the same time, which will mean an sync counter.